src/terrain/terrain.hpp

```diff
@@ -82,6 +84,18 @@ getter_low(Tile* t) {
     return t->get_grow_data_low();
 }
 
+template <class T>
+inline bool
+astar_compare(T* lhs, T* rhs) {
+    return lhs->get_total_predicted_cost() > rhs->get_total_predicted_cost();
+}
+
+template <class T>
+inline bool
+breadth_first_compare(T* lhs, T* rhs) {
+    return lhs->get_time_cots() > rhs->get_time_cots();
+}
+
```

Comment:
Alem

Am I going to have to explain translation units to you again, and how `#include` is a copy-paste?

If its an implementation detail and only used in the cpp file, put it in the cpp file and mark it inline. Then, it will be inlined.